### PR TITLE
Block access widening.

### PR DESCRIPTION
**Pull request by LambdAurora**
Access widening a bunch of block-related stuff transitively.

Based on fabric-transitive-access-wideners.

### CODE REVIEW

**Review by OroArmor — APPROVED** [review]
Works. However, I would leave the current file as is, and have the auto generated content be added to a second file that is gitignore d.

**Comment by LambdAurora** [comment]
> Works. However, I would leave the current file as is, and have the auto generated content be added to a second file that is gitignore d.

Pretty sure we can't have multiple access widener files per mod.

**Comment by OroArmor** [comment]
QMJ spec says otherwise

**Comment by TheGlitch76** [comment]
Loom does not currently support multiple AWs.

File: library/block/block_extensions/src/main/resources/quilt_block_extensions.accesswidener [hunk 1]
```diff
@@ -1,3 +1,104 @@
-accessWidener	v1	named
+accessWidener v2 named
+
 extendable  method    net/minecraft/block/AbstractBlock$Settings    <init>  (Lnet/minecraft/block/Material;Ljava/util/function/Function;)V
 extendable  method    net/minecraft/block/AbstractBlock$Settings    <init>  (Lnet/minecraft/block/Material;Lnet/minecraft/block/MapColor;)V
+
+### Generated access wideners below
```
Comment (TheGlitch76) [inline library/block/block_extensions/src/main/resources/quilt_block_extensions.accesswidener:6]: We intentionally avoid committing generated QMJs, but i suppose this is fine.

File: library/block/block_extensions/template.accesswidener [hunk 2]
```diff
@@ -0,0 +1,6 @@
+accessWidener v2 named
```
Comment (TheGlitch76) [inline library/block/block_extensions/template.accesswidener:1]: could we have a README.txt or something in this module explaining how to add things to the generated access widener?

**Review by TheGlitch76 — COMMENTED** [review]
Everything Access Widener in QSL
Everything Access Widener in QSL
Everything Access Widener in QSL

File: library/block/block_extensions/src/main/resources/quilt_block_extensions.accesswidener (hunk 1 above)
Comment (LambdAurora) [reply library/block/block_extensions/src/main/resources/quilt_block_extensions.accesswidener:6]: I tried to move the entire accesswidener file to the generated resources source set like for the QMJ
but loom was failing to read it sadly.

So I gave up on that one.

File: library/block/block_extensions/template.accesswidener (hunk 2 above)
Comment (LambdAurora) [reply library/block/block_extensions/template.accesswidener:1]: Added!

**Review by TheGlitch76 — APPROVED** [review]

**Comment by LemmaEOF** [comment]
Would it be possible to generate wideners for only `init` methods that *aren't* public?

**Comment by LambdAurora** [comment]
> Would it be possible to generate wideners for only `init` methods that _aren't_ public?

@LemmaEOF that's already what it does:

```java

					// Block constructor...
					if (method.name == "<init>" && Type.getArgumentTypes(method.desc).any { it.internalName == 'net/minecraft/block/AbstractBlock$Settings' }) {
						// ...and non-public
						if ((method.access & Opcodes.ACC_PUBLIC) == 0) {
							lines.add("transitive-accessible method $node.name <init> $method.desc")
						}
					}
```

**Comment by LemmaEOF** [comment]
Oh, huh. I didn't realize that many block ctors were non-public.

**Review by Leo40Git — APPROVED** [review]

**Pull request merged**